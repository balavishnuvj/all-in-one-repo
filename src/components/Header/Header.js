/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React from 'react';
import { Button } from 'antd';
import Link from '../Link';
import Navigation from '../Navigation';
import logoUrl from './logo-small.png';
import logoUrl2x from './logo-small@2x.png';

class Header extends React.Component {
  render() {
    return (
      <div className="root">
        <div className="container">
          <Navigation />
          <Button type="primary">Hello</Button>
          <Link className="brand" to="/">
            <img
              src={logoUrl}
              srcSet={`${logoUrl2x} 2x`}
              width="38"
              height="38"
              alt="React"
            />
            <span className="brandTxt">Your Company</span>
          </Link>
          <div className="banner">
            <h1 className="bannerTitle">React</h1>
            <p className="bannerDesc">Complex web apps made easy</p>
          </div>
        </div>
      </div>
    );
  }
}

export default Header;
