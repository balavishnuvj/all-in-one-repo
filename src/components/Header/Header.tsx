import * as React from 'react';
import {
  HeaderContainer,
} from './Header.styled';
import IogoSmall from './logo-small.png';

interface IProps {
  testingProps?: string;
}

// tslint:disable-next-line function-name
function Header(props: IProps) {
  const { testingProps } = props;
  return (
    <HeaderContainer>
      <span>
        <img src={IogoSmall} alt={'logo'} />
      </span>
      <span>
        Hello {testingProps} !
      </span>
    </HeaderContainer>
  );
}

export default Header;
