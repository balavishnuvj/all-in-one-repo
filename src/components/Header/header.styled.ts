import styled from 'styled-components';

export const HeaderContainer = styled.div`
    background: black;
    color: white;
    padding: 20px;
    display: flex;
    justify-content: space-between;
`;
