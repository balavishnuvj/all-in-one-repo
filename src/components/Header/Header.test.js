/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';
import createApolloClient from '../../utils/shared/apollo';

describe('Header', () => {
  test('renders children correctly', () => {
    const client = createApolloClient();
    const wrapper = renderer
      .create(
        <Header
          context={{
            insertCss: () => {},
            fetch: () => {},
            client,
            pathname: '',
          }}
        />,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
