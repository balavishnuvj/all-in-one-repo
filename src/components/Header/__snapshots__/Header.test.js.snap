// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders children correctly 1`] = `
<Header.tsx
  context={
    Object {
      "client": ApolloClient {
        "cache": InMemoryCache {
          "addTypename": true,
          "cacheKeyRoot": CacheKeyNode {
            "children": null,
            "key": null,
          },
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
        "clearStoreCallbacks": Array [],
        "clientAwareness": Object {},
        "defaultOptions": Object {},
        "disableNetworkFetches": true,
        "link": ApolloLink {
          "request": [Function],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "ssrMode": true,
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
        },
        "version": "2.4.12",
        "watchQuery": [Function],
      },
      "fetch": [Function],
      "insertCss": [Function],
      "pathname": "",
    }
  }
/>
`;
